matchms/similarity/CosineHungarian.py

```diff
@@ -69,15 +73,15 @@ def get_matching_pairs_matrix():
             matching_pairs_matrix:
                 Array of multiplied intensities between all matching peaks.
             """
-            if len(matching_pairs) == 0:
+            if matching_pairs is None:
                 return None, None, None
-            paired_peaks1 = list({x[0] for x in matching_pairs})
-            paired_peaks2 = list({x[1] for x in matching_pairs})
+            paired_peaks1 = list(set(matching_pairs[:, 0]))
+            paired_peaks2 = list(set(matching_pairs[:, 1]))
             matrix_size = (len(paired_peaks1), len(paired_peaks2))
             matching_pairs_matrix = numpy.ones(matrix_size)
-            for match in matching_pairs:
-                matching_pairs_matrix[paired_peaks1.index(match[0]),
-                                      paired_peaks2.index(match[1])] = 1 - match[2]
+            for i in range(matching_pairs.shape[0]):
+                matching_pairs_matrix[paired_peaks1.index(matching_pairs[i, 0]),
+                                      paired_peaks2.index(matching_pairs[i, 1])] = 1 - matching_pairs[i, 2]
             return paired_peaks1, paired_peaks2, matching_pairs_matrix
 
         def solve_hungarian():
```

Comment:
Line below has typo, `agorithm` should be `algorithm`

Reply:
done.